jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -209,10 +221,18 @@ public Response put(URL url, Request request) throws IOException {
    * @throws IOException if building the HTTP request fails.
    */
   public Response call(String httpMethod, URL url, Request request) throws IOException {
-    if (!isHttpsProtocol(url) && !enableHttpAndInsecureFailover) {
+    if (!isHttpsProtocol(url)) {
+      if (enableHttpAndInsecureFailover) {
+        return call(httpMethod, url, request, getHttpTransport(true));
+      }
       throw new SSLException("insecure HTTP connection not allowed: " + url);
     }
 
+    Optional<Response> optionalResponse = followFailoverHistory(httpMethod, url, request);
```

Comment:
I think the use of `optional` in the name `optionalReposonse` is unnecessary given the type information is already available.

Reply:
Are you suggesting we can just return `null` (instead of `Optional.empty()`) if there was previous recorded history?

Reply:
oh no, this is just naming.
```
Optional<Response> response = ....
```

Reply:
Sadly, the code below already took the name `response`, and I need to pick a different name for this.

Reply:
how about `responseFromFailoverPath`, or something that indicates the source?